### PR TITLE
Add Node controller unit tests

### PR DESCRIPTION
**Pull request by tpantelis**
Also fixed issues found and other minor changes. See individual commits for details,.

### CODE REVIEW

**Review by vthapar — APPROVED** [review]

File: pkg/globalnet/controllers/node_controller.go [hunk 1]
```diff
@@ -89,19 +90,25 @@ func (n *nodeController) process(from runtime.Object, numRequeues int, op syncer
...
-				_ = n.pool.Release(existingGlobalIP)
-				return nil, false
-			} else {
-				_ = n.pool.Release(existingGlobalIP)
-				return n.updateNodeAnnotation(node, "")
-			}
+		if existingGlobalIP := node.GetAnnotations()[constants.SmGlobalIP]; existingGlobalIP != "" {
+			cniIfaceIP := node.GetAnnotations()[constants.CNIInterfaceIP]
+
+			requeue := n.flushRulesAndReleaseIPs(node.Name, numRequeues, func(allocatedIPs []string) error {
```
Comment (sridhargaddam) [inline pkg/globalnet/controllers/node_controller.go]: We dont have to flush rules when the event corresponds to a different node as those rules would have been present on that OLD Gateway node and NOT on the current Active Gateway node. We just have to release the IP back to Pool and remove the annotation.

**Review by sridhargaddam — CHANGES_REQUESTED** [review]
Thanks @tpantelis for working on the unit tests. A small change is required.

File: pkg/globalnet/controllers/node_controller.go (hunk 1 above)
Comment (tpantelis) [reply pkg/globalnet/controllers/node_controller.go]: ahh - right

Comment (tpantelis) [reply pkg/globalnet/controllers/node_controller.go]: Removed the commit that changed it.

**Review by sridhargaddam — APPROVED** [review]

**Pull request merged**